### DISCUSSION TITLE
404 when saving information on the user directily after signup with email verification

### DISCUSSION THREAD

**Discussion by AdrianKrebs** [supabase/supabase#2927] [Questions]
I'm asking for the name of the user when signing up with email. The signup works fine, and I see the name and the user printed out (I added the log statements). However, the patch request to update the name on the user table returns a 404:
```
got user {id: "f79d6fce-6a19-4712-a8d5-00a2563eacf8", aud: "authenticated", role: "authenticated", email: "`xyz@email.com", phone: "", …}app_metadata: {provider: "email"}aud: "authenticated"confirmation_sent_at: "2021-08-19T11:56:42.870927667Z"created_at: "2021-08-19T11:56:42.867851Z"email: "xyz@email.com" id: "f79d6fce-6a19-4712-a8d5-00a2563eacf8"phone: ""role: "authenticated"updated_at: "2021-08-19T11:56:45.036149Z"user_metadata: null[[Prototype]]: Object
signup.js:32 

got name Adrian Krebs

PATCH https://nqltpulqdgngkiwpnjkj.supabase.co/rest/v1/users?id=eq.f79d6fce-6a19-4712-a8d5-00a2563eacf8 404
```

The user exists, so why is the update failing with a 404?

```
const handleSignup = async (e) => {
    e.preventDefault();

    setLoading(true);
    setMessage({});
    const { error, user } = await signUp({ email, password });
    if (error) {
      setMessage({ type: 'error', content: error.message });
    } else {
      if (user) {
        console.log('got user', user)
        console.log('got name', name)
        await updateUserName(user, name);
        setUser(user);
      } else {
        setMessage({
          type: 'note',
          content: 'Check your email for the confirmation link.'
        });
      }
    }
    setLoading(false);
  };

export const updateUserName = async (user, name) => {
  await supabase
    .from('users')
    .update({
      full_name: name
    })
    .eq('id', user.id);
};
```

**Comment by silentworks** [comment]
It's probably failing because you have row level security (RLS) set on that table. This would stop you from updating that table if the user is not logged in.

**Reply by Senninseyi** [reply]
> It's probably failing because you have row level security (RLS) set on that table. This would stop you from updating that table if the user is not logged in.

i dunno if the @AdrianKrebs  has fixed the situation but this is the solution to the problem

**Comment by CaseyG5** [comment]
Assuming you figured this out by now since it's been a few weeks.  But curious if you're getting the 404 when calling `updateUserName()`.  If so, are both `name` and `id` in your `users` table able to receive an update?  If one or both fields is empty, try `insert` or `upsert` instead.
